lib/xopts.ts

```diff
@@ -0,0 +1,74 @@
+import {Signer} from 'ethers';
+import {providers} from 'ethers';
+import {ReadOnlyContracts, ReadWriteContracts} from './contracts';
+import {Addresses, Deployments} from './addresses';
+import {ExchangeRate, Currency, MonetaryAmount} from './monetary';
+
+type SignerOrProvider = Signer | providers.Provider;
+
+interface Option<From extends Currency, To extends Currency> {
+  expiry: Date;
+  from: From;
+  to: To;
+  strikePrice: ExchangeRate<From, To>;
```

Comment:
Where would we input the amount of options/underlying? If this is an exchange rate, then the exchange rate is per unit of underlying to unit of collateral.

Reply:
I think this could be a method of the `Option` interface using the `strikePrice` to convert from/to the underlying.